### PR TITLE
[c++]Priority_task_chalti.cpp

### PR DESCRIPTION
**Pull request by chaltidutta**
My PR checks all the below points.

 - [x] Code is added in right directory(Folder).
 - [x] Brief Description of Solution is given.
 - [x] Time and Space Complexity is mentioned
 - [x]  Code is well indented.
 - [x]  Comments are clearly written wherever necessary
 - [x]  Code is successfully compiled and passes all test cases.

### CODE REVIEW

File: 4. Stack/priority_tasks/prioriy_task(bruteforce)_chalti.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+void solve(){
+    int n;
+    cin>>n;
+    int arr[n+n];
+    for(int i=0 ; i<n ; i++){
+        cin>>arr[i];
+        arr[n+i]=arr[i];
+    }
+    // int temp = -1;
```
Comment (anjali1903) [inline 4. Stack/priority_tasks/prioriy_task(bruteforce)_chalti.cpp]: Delete this line.

**Comment by anjali1903** [comment]
Your optimal approach would throw error(index out of bounds).

**Comment by chaltidutta** [comment]
> Your optimal approach would throw error(index out of bounds).

It works fine as well in my local ide

File: 4. Stack/priority_tasks/priority_task(most_optimal)_chalti.cpp [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    vector<int> temp(n);
+    for(int i=0 ; i<n ; i++){
+        temp[i]=-1; //declearing all the elements as -1
+    }
+    for(int i=0 ; i<(n+n) ; i++){
+        while(!stk.empty() && arr[i]>arr[stk.top()]){ //ckecking
+            temp[stk.top()]=arr[i]; //push the value of the arr into the vector
+            stk.pop(); 
+        }
+        stk.push(i); //lastly push the element
```
Comment (anjali1903) [inline 4. Stack/priority_tasks/priority_task(most_optimal)_chalti.cpp:43]: You're pushing i onto the stack which goes till 2*n but the size of temp is only n.

**Pull request closed without merging**